pinot-spi/src/main/java/org/apache/pinot/spi/utils/JsonUtils.java

```diff
@@ -719,4 +719,18 @@ private static void addFieldToPinotSchema(Schema pinotSchema, DataType dataType,
       }
     }
   }
+
+  public static List<Map<String, String>> flatten(String jsonString, JsonIndexConfig jsonIndexConfig) throws IOException {
+    JsonNode jsonNode;
+    try {
+      jsonNode = JsonUtils.stringToJsonNode(jsonString);
+    } catch (IOException e) {
+      if (jsonIndexConfig.getSkipInvalidJson()) {
+        return Collections.singletonList(Collections.singletonMap(VALUE_KEY, SKIPPED_VALUE_REPLACEMENT));
```

Comment:
why create this singleton at runtime? can't we make a static obj?

Reply:
good catch